crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -382,6 +390,157 @@ impl Persistence {
+                .await?
+        };
+
+        // Fetch the orders that were updated after the given block and were created or
+        // cancelled after the given timestamp.
+        let next_orders: HashMap<domain::OrderUid, model::order::Order> = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["open_orders_after"])
+                .start_timer();
+
+            database::orders::open_orders_after(&mut tx, &updated_order_uids, after_timestamp)
+                .map(|result| match result {
+                    Ok(order) => full_order_into_model_order(order)
+                        .map(|order| (domain::OrderUid(order.metadata.uid.0), order)),
+                    Err(err) => Err(anyhow::Error::from(err)),
+                })
+                .try_collect()
+                .await?
+        };
+
+        let next_order_uids = next_orders.keys().cloned().collect::<HashSet<_>>();
+
+        // Interactions table doesn't contain block number, so we need to update
+        // interactions for orders from the cache.
+        let current_order_interactions = {
+            let _timer = Metrics::get()
+                .database_queries
+                .with_label_values(&["read_interactions_for_orders"])
+                .start_timer();
+
+            let uids_to_update_interactions = current_orders
+                .orders
+                .keys()
+                .filter_map(|uid| (!next_order_uids.contains(uid)).then_some(ByteArray(uid.0)))
+                .collect::<Vec<_>>();
+            let interactions = database::orders::read_interactions_for_orders(
+                &mut tx,
+                &uids_to_update_interactions,
+            )
+            .await?;
+            db_interactions_to_model(&interactions)?
+        };
+
+        let all_order_uids = next_order_uids.iter().chain(current_orders.orders.keys());
+        // Fetch quotes for new orders and also update them for the orders from the
+        // cache since they could also be updated.
+        let updated_quotes = self.postgres.read_quotes(all_order_uids).await?;
+
+        let latest_settlement_block =
+            database::orders::latest_settlement_block(&mut tx).await? as u64;
+
+        Self::build_solvable_orders(
+            current_orders,
+            current_order_interactions,
+            next_orders,
+            updated_quotes,
+            latest_settlement_block,
+            min_valid_to,
+        )
+    }
+
+    fn build_solvable_orders(
+        current_orders: &boundary::SolvableOrders,
+        current_orders_interactions: HashMap<domain::OrderUid, model::order::Interactions>,
+        next_orders: HashMap<domain::OrderUid, model::order::Order>,
+        mut new_quotes: HashMap<domain::OrderUid, domain::Quote>,
+        latest_settlement_block: u64,
+        min_valid_to: u32,
+    ) -> anyhow::Result<boundary::SolvableOrders> {
+        let mut orders = current_orders.orders.clone();
```

Comment:
why not to pass the `current_orders` as mutable directly? so we avoid an internal clone

Reply:
Your suggestion could leave the cache in an unpredictable state in case something fails in the middle.

Reply:
I mean, if the orders in the cache were updated, but not the auction.